autoload/ale/handlers/rust.vim

```diff
@@ -50,7 +50,7 @@ function! ale#handlers#rust#HandleRustErrorsForFile(buffer, full_filename, lines
 
             if (
             \   l:span.is_primary
-            \   && (l:span_filename ==# l:filename || l:span_filename ==# '<anon>')
+            \   && (a:full_filename =~ (l:span.file_name . '$') || l:span_filename ==# '<anon>')
```

Comment:
Thank you for your fast reply!
`ale#path#IsBufferPath` does not work here. In my experience, `IsBufferPath`'s work is like this:

0. cwd = `/tmp/sample`, error file = `/tmp/sample/src/haserrors/mod.rs`, current file = `/tmp/sample/src/haserrors/mod.rs`
1. Get the file name of current buffer as `l:buffer_filename` ( = `/tmp/sample/src/haserrors/mod.rs`)
2. Apply `fnamemodify` to get enclosing directory of `l:buffer_filename` ( = `/tmp/sample/src/haserrors/` )
3. Concat the directory and error file name using `ale#path#GetAbsPath` ( = `/tmp/sample/src/haserrors/src/haserrors/mod.rs` )

`cargo` reports error with relative file path from the current directory, so the reported file name is `src/haserrors/mod.rs`.
(If you are in `src/haserrors`, the above solution works correctly)

Anyway, I will try to remove unused variables and add a test for the silent case.